### PR TITLE
Merge Porch into main branch

### PR DESCRIPTION
**Pull request by martinmaly**
- Simple package orchestration (#2663)
- oci-support feature branch (#2666)
- Remove Placeholder kptlib (#2665)
- Fix Porch Build (#2672)
- Correct Apache License Text (#2675)
- Implement Direct Package Fetch (#2667)
- Initial fn Interface (#2676)
- Upgrade go-containerregistry to v0.8.0 (#2674)
- git: handle resolution of per-package versions (#2681)
- PackageDraft interface should not inherit from Package (#2678)
- Minor fixups to better surface errors (#2680)
- Clean up Makefile (#2685)
- Render and Eval via Simple Memory FS (#2684)
- Working towards applying packages from a repository (#2686)
- Add priority and fairness to RBAC roles (#2692)
- Applying: publish MVP status (#2695)
- Fix Docker Build (#2696)
- Tidy Go Modules (#2699)
- Add Roundtrip Tests (#2703)
- Expose Renderer api with FileSystem abstraction (#2683)
- Create fix-all and verify-fix-all (#2702)
- Fix few file headers (#2707)
- Define a FunctionEvaluator gRPC Interface (#2700)
- Integrate porch with renderer interface (#2710)
- Introduce applyset abstraction (#2697)
- Git repository test (#2706)
- Create Kptfile in config/deploy (#2708)
- Update FunctionRunner Interface (#2714)
- Simple gRPC kpt function runner service (#2711)
- Use simple evaluator in tests (#2717)
- Fix os calls (#2718)
- git: create ApprovePackageRevision method, remove draft hack (#2712)
- Kpt location parsing (#2688)
- Use apply-setters directly (#2719)
- Upgrade kpt dependencies (#2727)
- Use Lower-Level Function Runner Interface (#2730)
- Use kyaml In-Memory Filesystem (#2731)
- Upgrade porch dependencies (#2726)
- Build Function Runner Docker Image (#2729)
- tests: Fix test flake from timestamp sanitization (#2713) (#2742)
- Create function-runner Deployment (#2733)
- Use kpt Renderer in Porch (#2735)
- RemoteRootSyncSet: use specified OCI image (#2722)
- RemoteRootSyncSet: apply changes with force (#2723)
- RemoteRootSyncSet: Remove hack where we set namespace (#2724)
- Create push-images target to push all our images (#2721)
- Create dockerfile / makefile / manifest for pushing controllers to cluster (#2694)
- Introduce mapTaskToMutation function (#2744)
- Fix kpt file system interactions (#2746)
- Pass Function Config to Eval (#2747)
- Connect to gRPC function runner (#2745)
- Fix crash (#2748)
- Fix clone directory rewriting (#2749)
- client generation: fix plural of packagerevisionresources (#2751)
- Update kpt dependencies (#2743)
- Streamline Docker Image Building (#2752)
- Run Porch With Consistent Working Dir (#2753)
- Add subpkgs test for porch (#2754)
- Basic e2e Test (#2755)
- Refactor Server Startup (#2770)
- Tidy in 1.17 compatible mode (#2769)
- Create Engine using Options (#2771)
- Support lazy credential resolution (#2772)
- Pass Context to OpenRepository (#2773)
- Use Context to Drive Server Shutdown (#2774)
- Clean up Required/Optional API Fields (#2778)
- Easier deployment onto GKE (#2776)
- Support Creating Porch Deployment Config (#2777)
- Update Porch Deployment and Instructions (#2782)
- Use controller-gen v0.8.0 (#2780)
- Set renderer when building CaDEngine (#2787)
- Fix missing error handling (#2784)
- Fix small typo (#2793)
- Don't call into kpt render if we don't have a package (#2788)
- Split git_test To Use Git Server (#2797)
- Add CreatePackageRevision Test (#2800)
- Cache apply-setters:v0.1 function (#2799)
- Return Git PackageRevisionResources Correctly (#2801)
- Make Git Server hostable in cluster (#2798)
- Add PackageRevision Approval API (#2810)
- Add Clone Package Test (#2807)
- Revert OCI and Location (#2814)

### CODE REVIEW

File: .github/workflows/porch.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  push:
+    paths-ignore:
+      - "docs/**"
+      - "site/**"
+  pull_request:
+    paths-ignore:
+      - "docs/**"
+      - "site/**"
+
+jobs:
```
Comment (droot) [inline .github/workflows/porch.yml:27]: Two things that will help us to analyze if we need to adjust the presubmit workflow (locally and in CI)
- How long does the porch specific tests take ? 
- What dependencies do a kpt contributor needs to install to run tests for porch

File: go.mod [hunk 2]
```diff
@@ -3,33 +3,33 @@ module github.com/GoogleContainerTools/kpt
...
 	gotest.tools v2.2.0+incompatible
-	k8s.io/api v0.22.3
-	k8s.io/apiextensions-apiserver v0.22.2
-	k8s.io/apimachinery v0.22.3
-	k8s.io/cli-runtime v0.22.2
-	k8s.io/client-go v0.22.2
-	k8s.io/klog/v2 v2.10.0
+	k8s.io/api v0.23.3
+	k8s.io/apiextensions-apiserver v0.23.3
+	k8s.io/apimachinery v0.23.3
```
Comment (droot) [inline go.mod:20]: This PR (https://github.com/GoogleContainerTools/kpt/pull/2796) also bumps up the k8s deps.

**Review by droot — COMMENTED** [review]
I reviewed the `non-porch` portion and seems good to me. Minor clarification on pre-submits. Might ping you offline.

File: go.mod (hunk 2 above)
Comment (martinmaly) [reply go.mod:20]: Good go know. Let's merge it then, without delay; it should then reduce the diff for this merge.

File: .github/workflows/porch.yml (hunk 1 above)
Comment (martinmaly) [reply .github/workflows/porch.yml:27]: Porch tests are about 5 mins (https://github.com/GoogleContainerTools/kpt/actions/runs/1867953046)

no dependencies needed.

The e2e tests will have mode where they run against a k8s cluster where they'd install Porch but that part is not yet implemented.

Comment (droot) [reply .github/workflows/porch.yml:27]: ok.

File: go.mod [hunk 3]
```diff
@@ -8,11 +8,11 @@ require (
...
-	github.com/otiai10/copy v1.6.0
+	github.com/otiai10/copy v1.7.0
 	github.com/philopon/go-toposort v0.0.0-20170620085441-9be86dbd762f
 	github.com/spf13/cobra v1.3.0
 	github.com/stretchr/testify v1.7.0
-	github.com/xlab/treeprint v0.0.0-20181112141820-a009c3971eca
+	github.com/xlab/treeprint v1.1.0
 	golang.org/x/mod v0.5.1
 	gotest.tools v2.2.0+incompatible
 	k8s.io/api v0.23.2
```
Comment (droot) [inline go.mod:18]: Thanks for aligning with `cli-utils'` version of k8s deps. This definitely made the merge easier.

**Review by droot — APPROVED** [review]

**Pull request merged**